asserts/ifacedecls_test.go

```diff
@@ -956,6 +985,79 @@ func (s *plugSlotRulesSuite) TestCompilePlugRuleConnectionConstraintsDeviceScope
 	}
 }
 
+func (s *plugSlotRulesSuite) TestCompilePlugRuleConnectionConstraintsSideArityConstraints(c *C) {
+	m, err := asserts.ParseHeaders([]byte(`iface:
+  allow-auto-connection: true`))
+	c.Assert(err, IsNil)
+
+	rule, err := asserts.CompilePlugRule("iface", m["iface"].(map[string]interface{}))
+	c.Assert(err, IsNil)
+
+	// defaults
+	c.Check(rule.AllowAutoConnection[0].SlotsPerPlug, Equals, asserts.SideArityConstraint{N: 1})
+	c.Check(rule.AllowAutoConnection[0].PlugsPerSlot.Any(), Equals, true)
+
+	c.Check(rule.AllowConnection[0].SlotsPerPlug.Any(), Equals, true)
+	c.Check(rule.AllowConnection[0].PlugsPerSlot.Any(), Equals, true)
+
+	// allow-connection => *
+	allowConnTests := []string{
+		`iface:
+  allow-connection:
+    slots-per-plug: 1
+    plugs-per-slot: 2`,
+		`iface:
+  allow-connection:
+    slots-per-plug: *
+    plugs-per-slot: 1`,
+		`iface:
+  allow-connection:
+    slots-per-plug: 2
+    plugs-per-slot: *`,
+	}
+
+	for _, t := range allowConnTests {
+		m, err = asserts.ParseHeaders([]byte(t))
+		c.Assert(err, IsNil)
+
+		rule, err = asserts.CompilePlugRule("iface", m["iface"].(map[string]interface{}))
+		c.Assert(err, IsNil)
+
+		c.Check(rule.AllowConnection[0].SlotsPerPlug.Any(), Equals, true)
+		c.Check(rule.AllowConnection[0].PlugsPerSlot.Any(), Equals, true)
+	}
+
+	// allow-auto-connection => *
+	allowAutoConnTests := []struct {
+		rule         string
+		slotsPerPlug int
+	}{
+		{`iface:
+  allow-auto-connection:
+    slots-per-plug: 1
+    plugs-per-slot: 2`, 1},
+		{`iface:
+  allow-auto-connection:
+    slots-per-plug: *
+    plugs-per-slot: 1`, -1},
+		{`iface:
+  allow-auto-connection:
+    slots-per-plug: 2
+    plugs-per-slot: *`, 1},
```

Comment:
I would again add a comment here that the 2 => 1 transformation is temporary and expected.

Reply:
I'll add a comment.